lib/indexer/indexer.js

```diff
@@ -127,7 +128,20 @@ class Indexer extends EventEmitter {
    */
 
   async close() {
-    return this.db.close();
+    await this.db.close();
+    return this.handleClose();
+  }
+
+  /**
+   * Close indexer. Unbind all events.
+   * @private
+   */
+
+  async handleClose() {
+    for (const [obj, event, listener] of this.bound)
+      obj.removeListener(event, listener);
```

Comment:
Is `removeAllListeners()` too extreme for this? Might be simpler.

https://nodejs.org/api/events.html#events_emitter_removealllisteners_eventname

Reply:
I don't immediately see a reason why not other than maybe it's preferable to use a lighter touch unless it's known why not. Mainly though I basically just took all these changes directly from bcoin's node implementation: https://github.com/bcoin-org/bcoin/blob/master/lib/node/node.js#L212